### DISCUSSION TITLE
Social plugin causing "Connection refused error"

### DISCUSSION THREAD

**Discussion by holgerflick** [squidfunk/mkdocs-material#6930] [Q&A] [answered]
Hi everybody,

I only have this issue on one of my two computers. Both are macOS and have the same config. At least I thought so. 

`mkdocs serve` does not work properly on the system that I installed recently. I could determine the `social` plugin as the cause of the error messages. I can ignore the error messages, but what makes this worse is that after the first successful creation on the next file change and rebuild the building process gets stuck and I have to restart `mkdocs serve` in the Terminal.

This is the error message I get:

`WARNING:root:Remote image could not been reached: http://127.0.0.1:8000/assets/images/social/blog/posts/14-results.png. Trying again with GET and disabling SSL verification. Attempt: 1. Trace: <urlopen error [Errno 61] Connection refused>
WARNING:root:Remote image could not been reached: http://127.0.0.1:8000/assets/images/social/blog/posts/14-results.png. Trying again with GET and disabling SSL verification. Attempt: 2. Trace: <urlopen error [Errno 61] Connection refused>`

I get this for all of the posts in my "first" blog. I have multiple blogs on the site. I also tried rearranging different plugins and also defining social after and before the blogs. No change. As said, it only happens to the blog in the `blog` directory.

I also tried several configurations for the plugin explicitly listing all directories. No success.

Any hints? I apologize if I missed any setting that is required for this to work, but the amount of configuration options makes it easy to miss something ...

**Comment by squidfunk** [comment]
Could you please provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/)? It is unclear whether this is an environment error or a bug in the social plugin. We need to see if we can reproduce it on our machines to be sure. Also note that the code base of the social plugin in the community edition and Insiders are different, as the plugin got a rewrite that was yet not released to the community edition. If you follow the reproduction steps, we'll get all information we need.

**Reply by holgerflick** [reply]
I would gladly provide a reproducible case, but it works on the other machine. So what would I provide to make it "reproduce" the problem as it is clearly the environment. Is there any way to "dump" my python and Homebrew config so that can be excluded as a cause for an error?

Both machines are set up to use the insider edition - at least I hope they do. Maybe that is exactly the mistake I made. Let me check that too...

**Reply by squidfunk** [reply]
Yes, I understand that it works on one and and doesn't on the other. Maybe mine is one that doesn't work as well given your configuration? We need to see what _might_ be the cause, so a reproduction that crashes on one machine is a start.

**Reply by holgerflick** [reply]
This might be the cause:
```
>> pip --version
pip 24.0 from /opt/homebrew/lib/python3.11/site-packages/pip (python 3.11)
>> pip3 --version
pip 24.0 from /opt/homebrew/lib/python3.12/site-packages/pip (python 3.12)
```

I really **dislike** how Python is handled on macOS. I clearly have two versions competing with each other here... The other Mac has pip and pip3 pointing to the same thing.

It's really a mess and I have no idea how to fix it as Homebrew and the preinstalled stuff in macOS are not in sync. I also cannot "start over" because of dependencies... **Sigh**

**Reply by holgerflick** [reply]
I was able to make `pip` point to `pip3` but that still won't fix the issue. Will try to build a minimal one.

**Reply by holgerflick** [reply]
After building a minimal test case I was able to determine that the `rss` plugin with the following settings was the culprit:

```
  - rss:
      match_path: blog/posts/.*
      date_from_meta:
        as_creation: date
      categories:
        - categories
        - tags
```

The error goes away and the recompilation does not freeze when I comment out those lines.

**Reply by holgerflick** [reply]
This is the "fix":
```
- rss:
      use_material_social_cards: false
      match_path: blog/posts/.*
      date_from_meta:
        as_creation: date
      categories:
        - categories
        - tags
 ```

It disables social cards in the RSS feed.

**Reply by squidfunk** [reply]
Could you please attach this reproduction here?

**Reply by squidfunk** [reply]
The thing that is weird here is that the social plugin does not fetch or request the image it generates. It fetches fonts from Google (which seems to work in your case, as social cards get generated, right?), but that's about it. This sounds to me that the RSS plugin could be doing something here that leads to the error. Thus it might be worth reporting upstream, if you can pinpoint the problem

**Reply by squidfunk** [reply, marked as answer]
[There it is](https://github.com/Guts/mkdocs-rss-plugin/blob/622655943443db9e0fe7e5c64a545095daa15812/mkdocs_rss_plugin/util.py#L650). The RSS plugin fetches the image and that seems to run into an error with SSL. Please report this upstream.

**Reply by holgerflick** [reply]
Also filed an issue with the project on GitHub best I can express it.
